test/test-https.js

```diff
@@ -51,9 +52,19 @@ tape('setup', function (t) {
 });
 
 tape('make request to https server', function (t) {
+	got('https://google.com', {
+		strictSSL: true
+	}, function (err, data) {
+		t.error(err);
+		t.ok(data);
+		t.end();
+	});
+});
+
+tape('make request to https server with ca', function (t) {
 	got(s.url, {
 		strictSSL: true,
-		ca: caRootCert,
+		agent: new https.Agent({ca: caRootCert}),
```

Comment:
Node [`https.js`](https://github.com/joyent/node/blob/v0.10.35-release/lib/https.js#L109) has this code to achieve this:

```js
  if (typeof options.agent === 'undefined') {
    if (typeof options.ca === 'undefined' &&
        typeof options.cert === 'undefined' &&
        typeof options.ciphers === 'undefined' &&
        typeof options.key === 'undefined' &&
        typeof options.passphrase === 'undefined' &&
        typeof options.pfx === 'undefined' &&
        typeof options.rejectUnauthorized === 'undefined') {
      options.agent = globalAgent;
    } else {
      options.agent = new Agent(options);
    }
  }
```

Should we port this logic as well? It's, kind of, sucks.